tests/e2escenarios/e2e-test.go

```diff
@@ -0,0 +1,274 @@
+			Expect(pvcs).To(BeEmpty())
+			pods := commonVar.CliRunner.GetAllPodNames(commonVar.Project)
+			Expect(pods).To(BeEmpty())
+
+			// "run odo deploy"
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-deploy.yaml"), path.Join(commonVar.Context, "devfile.yaml"))
+			helper.ReplaceString(filepath.Join(commonVar.Context, "devfile.yaml"), "nodejs-prj1-api-abhz", componentName)
+
+			stdout = helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
+			Expect(stdout).To(ContainSubstring("Your Devfile has been successfully deployed"))
+
+			// should deploy new changes
+			stdout = helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "nodejs", "Deploy"})
+
+			// start dev mode again
+			devSession, _, _, ports, err = helper.StartDevMode()
+			Expect(err).ToNot(HaveOccurred())
+
+			// making changes to the project again
+			helper.ReplaceString(filepath.Join(commonVar.Context, "server.js"), "from Node.js app v2", "from Node.js app v3")
+			_, _, err = devSession.WaitSync()
+			Expect(err).ToNot(HaveOccurred())
+			// "should update the changes"
+			checkIfDevEnvIsUp(ports["3000"], "Hello from Node.js app v3 Starter Application!")
+
+			// should list both dev,deploy
+			stdout = helper.Cmd("odo", "list").ShouldPass().Out()
+			helper.MatchAllInOutput(stdout, []string{componentName, "nodejs", "Dev", "Deploy"})
+
+			// "exit dev mode and run odo deploy"
+			devSession.Stop()
+
+			// "run odo deploy"
+			stdout = helper.Cmd("odo", "deploy").AddEnv("PODMAN_CMD=echo").ShouldPass().Out()
+			Expect(stdout).To(ContainSubstring("Your Devfile has been successfully deployed"))
+
+			// "run odo delete and check if the component is deleted"
+			command = []string{"odo", "delete", "component"}
+			_, err = helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+				helper.ExpectString(ctx, "Are you sure you want to delete \""+componentName+"\" and all its resources?")
+				helper.SendLine(ctx, "y")
+				helper.ExpectString(ctx, "successfully deleted")
+			})
+			Expect(err).To(BeNil())
+			Eventually(string(commonVar.CliRunner.Run(getDeployArgs...).Out.Contents()), 60, 3).ShouldNot(ContainSubstring(deploymentName))
+			Eventually(string(commonVar.CliRunner.Run(getSVCArgs...).Out.Contents()), 60, 3).ShouldNot(ContainSubstring(serviceName))
+		})
+	})
+
+	Context("starting with non-empty Directory", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), commonVar.Context)
+		})
+		It("should verify developer workflow from non-empty Directory", func() {
+			deploymentName := "my-component"
+			serviceName := "my-cs"
+			getDeployArgs := []string{"get", "deployment", "-n", commonVar.Project}
+			getSVCArgs := []string{"get", "svc", "-n", commonVar.Project}
```

Comment:
You can perhaps move these to the main `Context` since it's common for both the sub-contexts.

Reply:
Previously, Philippe recommended that we should define the these value close to tests where it's getting used. 
cc @feloy